### ISSUE TITLE
When showing / hiding columns cells become misaligned

### ISSUE THREAD

**Issue by rathboma** [olifolkerd/tabulator#3886]
Working on reproducing this, but here's the code and a gif that causes it:

Looks like when hiding columns, then combo showing/hiding columns it cases the next page in the vdom to become misaligned.

```js
 applyColumnChanges(columns) {
      if (!this.tabulator) return;

      this.tabulator.blockRedraw();
      
      
      columns.forEach(({name, filter}) => {
        if(filter) this.tabulator.showColumn(name)
        else this.tabulator.hideColumn(name)
      })

      this.tabulator.restoreRedraw();

      this.columnsWithFilterAndOrder = columns

      // this.tabulator.setData()
    }

```
![show-hide-misalignment](https://user-images.githubusercontent.com/279769/181836283-25049be3-eeef-4e48-93d9-5b4ff21f4857.gif)

**Comment by olifolkerd** [comment]
Is this using 5.3 and the virtual hoz renderer?

I don't think I tested that combo with the updated 5.3 redraw functionality. Should be a fairly quick fix, I will see if I can get that in the next patch release

Cheers

Oli :)

**Comment by rathboma** [comment]
Yep!

Ok, thanks Oli

**Comment by rathboma** [comment]
@olifolkerd so this is on 5.1 onwards.

I'm actually not sure if the virtual horizontal renderer is used because I'm using a frozen column. Does that disable the virtual renderer?

**Comment by olifolkerd** [comment]
Hey @rathboma 

You cant use horizontal columns and the horizontal vDOM at the same time, they adjust the same margins/paddings each which will result in them fighting against each other and result in misalignment of columns,

Could you confirm this happens if you dont have frozen columns enabled? if this only occurs with frozen columns then this isnt a bug but a known compatibility limitation (included in the documentation) of the current system and would need to be scheduled for an update.

Let me know either way,

Cheers

Oli :)

**Comment by rathboma** [comment]
let me check.

**Comment by rathboma** [comment]
@olifolkerd this **does** happen without any frozen columns. Simply having the virtual horizontal DOM is enough to trigger it

**Comment by rathboma** [comment]
If I force a redraw `this.tabulator.redraw(true)` that solves it but makes the UI more laggy

**Comment by olifolkerd** [comment]
Do you have a minimal table setup I could use to test this?

Is this when you are using the blockRedraw function?

**Comment by rathboma** [comment]
I'll try to reproduce it for you

**Comment by olifolkerd** [comment]
Hey @rathboma 

I have pushed a fix for this to the 5.4 branch that i am hoping to release in the next week or so.

Cheers

Oli :)
